incomprehensible error message from program OK with gfortran and ifx

The program and the last few lines of its lfortran output:
```
(lf2) john:~/Test/Glist$ cat explogtest.f90
program explogtest
  ! Which is faster: exp or log?
  integer:: n, k = 8
  real:: x,dx,y,z,t1,t2
  print "(A,I0)",'k = ',k
  dx = 1e-6
  call cpu_time(t1)
  x = 0d0
  do n = 1,10**k
     x = x+dx
     z = exp(1d0/x)
  end do
  call cpu_time(t2)
  print "(2(A,F0.3))",'after 10**k exp, abs(t2-t1) = ',abs(t2-t1),' s; z=',z
  x = 0d0
  do n = 1,10**k
     x = x+dx
     z = log(1d0/x)
  end do
  call cpu_time(t1)
  print "(2(A,F0.3))",'after 10**k log, abs(t2-t1) = ',abs(t2-t1),' s; z=',z
end program explogtest

code generation error: asr_to_llvm: module failed verification. Error:
Call parameter type does not match function signature!
  %t1 = alloca float, align 4
 double*  call void @__module_lfortran_intrinsic_math_cpu_time(float* %t1)
Call parameter type does not match function signature!
  %t2 = alloca float, align 4
 double*  call void @__module_lfortran_intrinsic_math_cpu_time(float* %t2)
Call parameter type does not match function signature!
  %t1 = alloca float, align 4
 double*  call void @__module_lfortran_intrinsic_math_cpu_time(float* %t1)
```
Output of the same program from gfortran (ifx was very similar)
```
k = 8
after 10**k exp, abs(t2-t1) = .899 s; z=1.032
after 10**k log, abs(t2-t1) = .915 s; z=-3.466
```

Pranavchiku's suggestion failed. I am not familiar with ```conda``` but managed to use it. I had never heard of ```git```. When I tried ```gh``` I got
```
(lf2) john:~$ gh pr checkout 3920
Command 'gh' not found, but can be installed with:
sudo snap install gh       # version 2.6.0-15-g1a10fd5a, or
sudo apt  install gh       # version 2.4.0+dfsg1-2
sudo apt  install gitsome  # version 0.8.0+ds-6ubuntu1
See 'snap info gh' for additional versions.
(lf2) 
```
so that although the ```conda create ...``` command seemed to work, downloading and installing various packages, 
```make toml``` and ```conda activate lf``` did not give me a working ```lfortran```. Evidence:
```
 (lf2) john:~$ make toml
make: *** No rule to make target 'toml'.  Stop.
(lf2) john:~$ conda activate lf
(lf) john:~$ lfortran explogtest.f90
Command 'lfortran' not found, did you mean:
  command 'gfortran' from deb gfortran (4:11.2.0-1ubuntu1)
Try: sudo apt install <deb name>
(lf) john:~$ 
```
So I deactivated lf.

--- a/integration_tests/CMakeLists.txt
+++ b/integration_tests/CMakeLists.txt
@@ -743,6 +743,7 @@ RUN(NAME intrinsics_209 LABELS gfortran llvm llvm_wasm) # atand
 RUN(NAME intrinsics_210 LABELS gfortran llvm llvm_wasm) # bessel_jn
 RUN(NAME intrinsics_211 LABELS gfortran llvm llvm_wasm) # bessel_yn
 RUN(NAME intrinsics_212 LABELS gfortran llvm) # specific type intrinsics
+RUN(NAME intrinsics_213 LABELS gfortran llvm) # cpu_time
 
 RUN(NAME parameter_01 LABELS gfortran)
 RUN(NAME parameter_02 LABELS gfortran)

--- a/integration_tests/intrinsics_213.f90
+++ b/integration_tests/intrinsics_213.f90
@@ -1,0 +1,25 @@
+program intrinsics_213
+  ! Which is faster: exp or log?
+  integer:: n, k = 8
+  real:: x,dx,y,z,t1,t2
+  print "(A,I0)",'k = ',k
+  dx = 1e-6
+  call cpu_time(t1)
+  x = 0d0
+  do n = 1,10**k
+     x = x+dx
+     z = exp(1d0/x)
+  end do
+  call cpu_time(t2)
+  print "(2(A,F0.3))",'after 10**k exp, abs(t2-t1) = ',abs(t2-t1),' s; z=',z
+  if (abs(z - 1.03174341) > 1e-8) error stop 
+  x = 0d0
+  do n = 1,10**k
+     x = x+dx
+     z = log(1d0/x)
+  end do
+  call cpu_time(t1)
+  print "(2(A,F0.3))",'after 10**k log, abs(t2-t1) = ',abs(t2-t1),' s; z=',z
+  if (abs(z - (-3.46573591)) > 1e-8) error stop
+end program intrinsics_213
+

--- a/src/libasr/runtime/lfortran_intrinsics.c
+++ b/src/libasr/runtime/lfortran_intrinsics.c
@@ -1882,8 +1882,12 @@ LFORTRAN_API int64_t _lfortran_ibits64(int64_t i, int32_t pos, int32_t len) {
 
 // cpu_time  -------------------------------------------------------------------
 
-LFORTRAN_API void _lfortran_cpu_time(double *t) {
+LFORTRAN_API void _lfortran_d_cpu_time(double *t) {
     *t = ((double) clock()) / CLOCKS_PER_SEC;
+}
+
+LFORTRAN_API void _lfortran_s_cpu_time(float *t) {
+    *t = ((float) clock()) / CLOCKS_PER_SEC;
 }
 
 // system_time -----------------------------------------------------------------

--- a/src/runtime/impure/lfortran_intrinsic_math.f90
+++ b/src/runtime/impure/lfortran_intrinsic_math.f90
@@ -15,19 +15,34 @@ interface random_number
     module procedure sp_rand_num, dp_rand_num
 end interface
 
+interface cpu_time
+    module procedure sp_cpu_time, dp_cpu_time
+end interface
+
 contains
 
 ! cpu_time ---------------------------------------------------------------------
 
-pure subroutine cpu_time(t)
+pure subroutine sp_cpu_time(t)
 real(dp), intent(out) :: t
 interface
-    pure subroutine c_cpu_time(t) bind(c, name="_lfortran_cpu_time")
+    pure subroutine c_d_cpu_time(t) bind(c, name="_lfortran_d_cpu_time")
     import :: c_double
     real(c_double), intent(out) :: t
     end subroutine
 end interface
-call c_cpu_time(t)
+call c_d_cpu_time(t)
+end subroutine
+
+pure subroutine dp_cpu_time(t)
+real(sp), intent(out) :: t
+interface
+    pure subroutine c_s_cpu_time(t) bind(c, name="_lfortran_s_cpu_time")
+    import :: c_float
+    real(c_float), intent(out) :: t
+    end subroutine
+end interface
+call c_s_cpu_time(t)
 end subroutine
 
 ! system_clock------------------------------------------------------------------

--- a/tests/reference/asr-matmul_01-7b0b0c2.json
+++ b/tests/reference/asr-matmul_01-7b0b0c2.json
@@ -6,7 +6,7 @@
     "outfile": null,
     "outfile_hash": null,
     "stdout": "asr-matmul_01-7b0b0c2.stdout",
-    "stdout_hash": "0a6f092a3f8fe441b02c52d8fc1124a7409ab1c12cd3d1adbe23077b",
+    "stdout_hash": "407f8243e46c943a1cf07479aefc7a86e71be8fb256be4252a16cd6e",
     "stderr": null,
     "stderr_hash": null,
     "returncode": 0

--- a/tests/reference/asr-matmul_01-7b0b0c2.stdout
+++ b/tests/reference/asr-matmul_01-7b0b0c2.stdout
@@ -123,6 +123,16 @@
                                     cpu_time
                                     Private
                                 ),
+                            cpu_time@sp_cpu_time:
+                                (ExternalSymbol
+                                    6
+                                    cpu_time@sp_cpu_time
+                                    8 sp_cpu_time
+                                    lfortran_intrinsic_math
+                                    []
+                                    sp_cpu_time
+                                    Private
+                                ),
                             dp:
                                 (Variable
                                     6
@@ -497,8 +507,8 @@
                         ()
                     )
                     (SubroutineCall
+                        6 cpu_time@sp_cpu_time
                         6 cpu_time
-                        ()
                         [((Var 6 t1))]
                         ()
                     )
@@ -532,8 +542,8 @@
                         []
                     )
                     (SubroutineCall
+                        6 cpu_time@sp_cpu_time
                         6 cpu_time
-                        ()
                         [((Var 6 t2))]
                         ()
                     )
@@ -712,8 +722,8 @@
                         ()
                     )
                     (SubroutineCall
+                        6 cpu_time@sp_cpu_time
                         6 cpu_time
-                        ()
                         [((Var 6 t1))]
                         ()
                     )
@@ -734,8 +744,8 @@
                         []
                     )
                     (SubroutineCall
+                        6 cpu_time@sp_cpu_time
                         6 cpu_time
-                        ()
                         [((Var 6 t2))]
                         ()
                     )
@@ -976,8 +986,8 @@
                         ()
                     )
                     (SubroutineCall
+                        6 cpu_time@sp_cpu_time
                         6 cpu_time
-                        ()
                         [((Var 6 t1))]
                         ()
                     )
@@ -998,8 +1008,8 @@
                         []
                     )
                     (SubroutineCall
+                        6 cpu_time@sp_cpu_time
                         6 cpu_time
-                        ()
                         [((Var 6 t2))]
                         ()
                     )

